examples/Cassie/run_controller_switch.cc

```diff
@@ -1,55 +1,136 @@
 
+using drake::systems::TriggerType;
 using drake::systems::lcm::LcmPublisherSystem;
+using drake::systems::lcm::TriggerTypeSet;
 
-DEFINE_double(end_time, std::numeric_limits<double>::infinity(),
-              "End time of simulation");
-DEFINE_double(publish_rate, 1000, "Publishing frequency (Hz)");
+DEFINE_string(channel_x, "CASSIE_STATE",
+              "The name of the channel which receives state");
 DEFINE_string(controller_channel, "PD_CONTROLLER",
               "The name of the lcm channel that dispatcher_in listens to");
+DEFINE_int32(n_fsm_period, -1,
+             "the number of state-switch after the start of the"
+             "simulator containing this diagram"
+             "Negative value means no time delay");
+DEFINE_double(fsm_period, -1.0,
+              "has to be the same as `duration_per_state` in "
+              "`TimeBasedFiniteStateMachine`");
+DEFINE_double(fsm_offset, 0.0,
+              "has to be the same as `time_shift` in "
+              "`TimeBasedFiniteStateMachine`");
+
+/// This program is a one-time-use switch that tells dispatcher_robot_in which
+/// channel to listen to. It publishes the lcm message, PublishMessageType,
+/// which contains a string of the channel name.
 
-/// This diagram publishes a string which tells dispatcher_robot_in which
-/// channel to listen to.
+/// The program is extended so that it could be used with
+/// `TimeBasedFiniteStateMachine`. More specifically, the string assigment
+/// starts when `TimeBasedFiniteStateMachine` switches to a new state (because
+/// we don't want to switch to a new controller when the new controller is in a
+/// middle of a discrete state). This requires that the users provide two (or
+/// three) more arguments to the constructors besides `controller_channel`:
+///   @param n_fsm_period, the number of state-switch after the start of the
+///   simulator containing this diagram
+///   @param fsm_period, has to be the same as `duration_per_state` in
+///   `TimeBasedFiniteStateMachine`
+///   @param fsm_offset, has to be the same as `time_shift` in
+///   `TimeBasedFiniteStateMachine`
+///
+/// Let t0 be the time of the simulator/robot when we started running the
+/// diagram containing this class. And let t_current be the current diagram
+/// time. This class outputs the channel name only when
+///   t_current >= (floor(t0/fsm_period) + n_fsm_period) * fsm_period +
+///   fsm_offset.
+/// That is, this thread starts outputting the new channel name when
+/// `TimeBasedFiniteStateMachine` switches to a new state for the
+/// `n_fsm_period`-th times after it starts running.
 
 int do_main(int argc, char* argv[]) {
   gflags::ParseCommandLineFlags(&argc, &argv, true);
 
+  // Ensure that if (n_fsm_period >= 0), then (period > 0).
+  DRAKE_DEMAND((FLAGS_n_fsm_period < 0) || (FLAGS_fsm_period > 0));
+  // offset has to be positive
+  DRAKE_DEMAND(FLAGS_fsm_offset >= 0);
+
+  // Parameters
+  using PublishMessageType = dairlib::lcmt_controller_switch;
```

Comment:
These `using` definitions, mid-function, hurt readability. Are they really necessary?

Reply:
It's not necessary. I wrote it like because I was thinking of generalizing the code.